### ISSUE TITLE
General Retrospective for July 2024 Releases

### ISSUE THREAD

**Issue by smlambert** [adoptium/temurin#50]
**Summary**

A retrospective for all efforts surrounding the titular releases. 

All community members are welcome to contribute to the agenda via comments below.

This will be a virtual meeting after the release, with at least a week of notice in the #release Slack channel.

On the day of the meeting we'll review the agenda and add a list of actions at the end.

Invited: Everyone.

**Time, Date, and URL**

Time:  3pm (GMT+1, UK), 10am (GMT -4, Ontario)
Date:  Wednesday, 31st of July
URL: https://meet.google.com/tny-iobv-omt
Phone: +44 20 3937 3489 and enter this PIN: 698 738 573#
To view more phone numbers, click this link: https://tel.meet/tny-iobv-omt?hs=5

**Details**

Retrospective Owner Tasks (in order):

- [x] Post retro URL in \#Release around the start of the new release.
- [x] Wait until most builds are released, with no signs of a respin.
- [x] Announce the retrospective's date + time on \#Release a week in advance.
- [x] Host the retrospective:
  - Go through the agenda.
  - Create a list of actions.
- [x] Process each action:
  - Create a "WIP" issue including the source comment.
  - Add the issue to the current iteration.
  - Add an issue link to the action list.
- [x] Create a new retrospective issue for the next release.
- [x] Set a calendar reminder so you remember to do step 1 before the next release.
- [x] Close this issue.

**TLDR**

Add proposed agenda items as comments below.

**Comment by smlambert** [comment]
Would be handy to link to the [release pipeline generator job](https://ci.adoptium.net/job/build-scripts/job/utils/job/release-pipeline-generator/ ) in the Checklist (instead of having to go to release doc to find it)

**Comment by smlambert** [comment]
Update release status template to include aarch64_alpine-linux where applicable

**Comment by smlambert** [comment]
Update code freeze message from 
```
With under a week to go until releases, we are entering a lockdown period for the master branches in the following repositories: github-release-scripts, containers, installer, and mirror-scripts. The temurin-build, ci-jenkins-pipelines and jenkins-helper master branches are still open for work, however the release branch will be under code-freeze management.
```

as it is not a 'lockdown period for master branches', but for the release branches.

**Comment by andrew-m-leonard** [comment]
It is not possible to dryrun jdk22u, as there is no jdk-22.0.2 upstream tags, and I don't feel we should artificially create a arbitrary "tag".
I think we should probably enable for these Oracle managed releases, to be able to launch a release-pipeline "Release" type build using the HEAD "dev" branch rather than a tag...?

**Comment by andrew-m-leonard** [comment]
> It is not possible to dryrun jdk22u, as there is no jdk-22.0.2 upstream tags, and I don't feel we should artificially create a arbitrary "tag". I think we should probably enable for these Oracle managed releases, to be able to launch a release-pipeline "Release" type build using the HEAD "dev" branch rather than a tag...?

Interesting, I’ve just manually kicked off a [release-openjdk22-pipeline](https://ci.adoptium.net/job/build-scripts/job/release-openjdk22-pipeline/22/) and it’s actually building HEAD of “dev” as the latest ga tag which is 22.0.1+8-ga
I was expecting the testenv check to fail…but it hasn’t I think because scm_ref is “”
So that will probably do, as it is building the very latest jdk-22.0.2 HEAD commits, but under a version string of jdk-22.0.1+8, which is probably fine as a dryrun

**Comment by smlambert** [comment]
I did not get https://github.com/adoptium/aqa-tests/pull/5456 merged until after pipelines triggered, some tests may have been triggered without the excludes in place.  Reruns should pick up the changes.

**Comment by jiekang** [comment]
Some test jobs failed as the test gathering process was sent to a machine that could not complete the job. We should have worker labels on nodes that are safe for these jobs.

Text below from: 
https://github.com/adoptium/aqa-tests/issues/5444#issuecomment-2233270448

### x86-64_windows 21

**extended.perf**

Failed when collecting artifacts via machine test-sxa-armv7l-ubuntu2004-odroid-2
```
[2024-07-17T06:00:38.100Z] Running on test-sxa-armv7l-ubuntu2004-odroid-2 in /ssd/jenkins/workspace/Test_openjdk21_hs_sanity.perf_x86-64_windows
...
[2024-07-17T06:04:49.333Z] Exception: hudson.remoting.RequestAbortedException: java.nio.channels.ClosedChannelException
```
https://ci.adoptium.net/job/Test_openjdk21_hs_sanity.perf_x86-64_windows/159/consoleText

**Comment by smlambert** [comment]
Scorecards make assumption that a -ga tag has dropped.  There has been no -ga tag for jdk8u422, https://github.com/adoptium/jdk8u/tags, so our calculations / scores won't get generated (and when they do, we will have to adjust based on a mailing list email or when we finally decided to trigger off of jdk8u422-b05

**Comment by tellison** [comment]
The marketplace data repository had it's update action suspended due to inactivity in the repository.
<img width="1097" alt="Screenshot 2024-07-20 at 08 10 04" src="https://github.com/user-attachments/assets/13d4888e-c260-4d23-92bf-1c5b01dffbd7">

Presumably this was notified by e-mail to some group (at Eclipse?) - it was only noticed by the project by chance. We need to receive such notifications and/or configure the repo to keep the actions alive longer.

**Comment by steelhead31** [comment]
Solaris AQA Pipeline failed due to issues with USE\_TESTENV\_PROPERTIES, the shell script it calls, does not work on solaris   
  
[https://github.com/adoptium/aqa-tests/blob/master/scripts/testenv/testenvSettings.sh](https://github.com/adoptium/aqa-tests/blob/master/scripts/testenv/testenvSettings.sh)

Falls out with the IF statement on line 5

**Comment by smlambert** [comment]
Update checklist to use correct link for homebrew cask https://github.com/Homebrew/homebrew-cask/blob/master/Casks/t/temurin.rb

**Comment by smlambert** [comment]
Same issue as previous releases, https://github.com/adoptium/temurin/issues/28#issuecomment-2077153603, where Release Notes do not show up on website even after a day of them being published to the binaries repo, https://adoptium.net/temurin/release-notes/?version=jdk-21.0.4+7 
<img width="708" alt="Screenshot 2024-07-23 at 1 14 14 PM" src="https://github.com/user-attachments/assets/d8fe1ef5-c325-4b0d-a903-ac615024c5ce">

**Comment by smlambert** [comment]
There are 2 mentions in checklist to check for JDK8 arm32 tags, one refers to Mercurial, is it outdated and if so, should be removed
See the item that reads: "Check for presence of jdk8u aarch32 GA tag and mirror it [Mercurial repo](https://hg.openjdk.java.net/aarch32-port/jdk8u) - [Mirror job](https://ci.adoptopenjdk.net/view/git-mirrors/job/git-mirrors/job/adoptium/job/git-hg-aarch32-jdk8u/)"

**Comment by smlambert** [comment]
Update the link that is listed in this checklist item, since it doesn't exist.

"Consider updating the API as required via the relevant parts of [the Adoptium API model constants](https://github.com/adoptium/api.adoptium.net/blob/main/adoptium-models-parent/adoptium-api-v3-models/src/main/kotlin/net/adoptium/api/v3/models/Versions.kt). "

**Comment by smlambert** [comment]
> We should re-order the pipeline such that signInstaller, gpgSign and verifySigning stages

+1 as this would also save resources to not trigger testing at all in the cases where those stages fail

**Comment by andrew-m-leonard** [comment]
The generated dry-run Publish job link for "Point release" re-builds, does not add the adopt_build_number to the TAG parameter

**Comment by smlambert** [comment]
Consider just adding all of the upcoming releases to the release calendar in 1 go and removing this item from the release checklist

Ensure the [adoptium calendar](https://calendar.google.com/calendar/u/0/embed?src=c_56d7263c0ceda87a1678f6144426f23fb53721480b5ff71b073afb51091e5492@group.calendar.google.com) is updated for the next cycle at a minimum

**Comment by andrew-m-leonard** [comment]
Mac homebrew cask updates process is not clear, especially for ".1" re-build updates, eg.https://github.com/Homebrew/homebrew-cask/pull/180691
Documentation not great if you've not done it before........?

**Comment by smlambert** [comment]
Notes from aqa-tests triage:

JDK8
- x64 Solaris: 2 SSL testcase failures, javax/net/ssl/ServerName/SSLEngineExplorerMatchedSNI.java & sun/management/jmxremote/bootstrap/SSLConfigFilePermissionTest.sh suspect due to new machine config, consider non-blocking
- arm32 linux and win32: Exclusion needed for compiler/loopopts/TestUnrollLimitPreciseType.java#test1 on this platform, using unrecognized VM option
- Decide what to do with TestTLS12 that fails on Ubuntu 24.04 nodes https://github.com/adoptium/aqa-tests/issues/5420

JDK17
- CheckCICompilerCount and GetXSpace fail and need investigation exclusion, non-blocking on arm_linux

JDK22
- x64 Windows: com/sun/net/httpserver/simpleserver/CommandLinePositiveTest.java and simpleserver/jwebserver/CommandLinePositiveTest.java are problematic for JDK22 and JDK21, address in use failures
- aarch64 Linux, 2 MBCS targets need fixing or exclusion, non-blocking
- riscv64: Decide if we should exclude SuspendResume1 since it can only pass on certain nodes

**Comment by smlambert** [comment]
The new reduced size rerun feature in aqa-tests is great, really reduced time it took to triage tests.

**Comment by smlambert** [comment]
Was great that @steelhead31 did all the installer items in the checklist.  Wondering if there will be updates to that process before October.

**Comment by sophia-guo** [comment]
Should we add the publish aqa-tests result taps tar to the checklist? Also somewhere to add the documents of how to do this?

**Comment by adamfarley** [comment]
### Actions

**Adam**
- [ ] These simple changes in a single PR:
  - https://github.com/adoptium/temurin/issues/50#issuecomment-2223156811
  - https://github.com/adoptium/temurin/issues/50#issuecomment-2223188669
  - https://github.com/adoptium/temurin/issues/50#issuecomment-2245724474
  - https://github.com/adoptium/temurin/issues/50#issuecomment-2245813824
  - Remove this line: https://github.com/adoptium/temurin/issues/50#issuecomment-2247888699
- [ ] Link issue already raised for: https://github.com/adoptium/temurin/issues/50#issuecomment-2245803555
- [ ] Raise eclipse issue and add action to release checklist: https://github.com/adoptium/temurin/issues/50#issuecomment-2240975680
- [ ] Raise issue to make sure that these are sent to a particular label (whatever the label ends up being) https://github.com/adoptium/temurin/issues/50#issuecomment-2233343598
- [ ] Raise and assign issue to Stewart for sorting out access to release calendar: https://github.com/adoptium/temurin/issues/50#issuecomment-2251527605
- [ ] Raise infra issue for the Solaris line: https://github.com/adoptium/temurin/issues/50#issuecomment-2253696539
- [ ] Raise AQA issue for CheckCICompilerCount and GetXSpace: https://github.com/adoptium/temurin/issues/50#issuecomment-2253696539
- [ ] x64 windows: raise issue, assign to scott: https://github.com/adoptium/temurin/issues/50#issuecomment-2253696539

**Andrew**
- [ ] https://github.com/adoptium/temurin/issues/50#issuecomment-2223258453
- [ ] Document this process (also, see the comment below it too): https://github.com/adoptium/temurin/issues/50#issuecomment-2225222776
- [ ] Raise issue: https://github.com/adoptium/temurin/issues/50#issuecomment-2250850730
- [ ] Add clarification (maybe just "has it already been done before") : https://github.com/adoptium/temurin/issues/50#issuecomment-2252399360

**Shelley**
- [ ] Create short-form release checklist for personal use (and possible later contribution).
- [ ] Exclude Arm32 tests mentioned here: https://github.com/adoptium/temurin/issues/50#issuecomment-2253696539
- [ ] Raise issue for this: https://github.com/adoptium/temurin/issues/50#issuecomment-2244532293

**Sophia**
- [ ] https://github.com/adoptium/temurin/issues/50#issuecomment-2260717363

**Other notes**
Already fixed: https://github.com/adoptium/temurin/issues/50#issuecomment-2237493875
